Source/DafnyCore/AST/Grammar/ProgramParser.cs

```diff
@@ -102,6 +104,23 @@ public class ProgramParser {
 
     ShowWarningsForIncludeCycles(program);
 
+    compilation.TranslationRecord = new TranslationRecord();
```

Comment:
Fair, I put it here because it felt like initialization of `CompilationData`, but I moved it to `SynchronousCliCompilation` and it seems to fit nicely there.

Reply:
Would moving it to ExecutionBackend also work? I see it's used around there and it doesn't seem useful earlier on.